src/passes/rejectUnsupportedFeatures.ts

```diff
@@ -174,6 +174,31 @@ export class RejectUnsupportedFeatures extends ASTMapper {
     } else if (node.kind === FunctionKind.Receive) {
       throw new WillNotSupportError(`Receive functions are not supported`, node);
     }
+
+    //Checking if the function defination is constructor and
+    // if it is, checking if 'this' keyword is used to call
+    // the external/public functions of the contract
+
+    if (node.kind === FunctionKind.Constructor) {
+      const nodesWithThisIdentifier = node.vBody
+        ?.getChildren()
+        .filter((childNode, _) => childNode instanceof Identifier && childNode.raw.name === 'this');
```

Comment:
Nitpick. childnode.raw.name => childnode.name.

Reply:
Could we also have all this logic wrapped up in a single function, for readability?

Reply:
done